src/journal/JournalRecorder.cc

```diff
@@ -406,4 +422,13 @@ void JournalRecorder::handle_overflow(ObjectRecorder *object_recorder) {
   close_and_advance_object_set(object_number / splay_width);
 }
 
+JournalRecorder::Lockers JournalRecorder::lock_object_recorders() {
+  Lockers lockers;
+  lockers.reserve(m_object_ptrs.size());
+  for (auto& lock : m_object_locks) {
+    lockers.emplace_back(lock);
```

Comment:
I think you need `lockers.emplace_back(Mutex::Locker(lock));` here.

Reply:
or probably just `lockers.emplace_back({lock});` will work too.

Reply:
Ah, it was a wrong advise, I thought it was `push_back`, not `emplace_back`, and forgot that `m_object_locks` is an array of shared pointers.

Anyway, it seems a little complicated. I think the following would work:
```
typedef std::vector<std::unique_ptr<Mutex::Locker>> Lockers;
```
and in `JournalRecorder::lock_object_recorders()` use:
```
lockers.push_back(std::make_unique<Mutex::Locker>(*lock));
```

But maybe @dillaman has a better suggestion.

Reply:
Yeah, definitely non-trivial. If you want, just assign the tracker ticket to me and I'll take care of it.